### PR TITLE
[DOC]: Improve documentation

### PR DESCRIPTION
**Pull request by synchon**
* [x] description of feature/fix
* [x] add an entry for the latest changes

This PR sets line length for `.rst` files to `80`, improves general prose and formatting along with code block updates for docs. Also, runs `black=23.x.x` for the codebase.

### CODE REVIEW

File: docs/extending/datagrabber.rst [hunk 1]
```diff
@@ -272,152 +272,172 @@ And we can create our datagrabber:
...
-However, we need to remember that for session *ses-03* there is no BOLD data.
+The second method, ``get_elements``, needs to return a list of all the elements
+in the dataset. In this case, we know that the dataset contains 3 subjects and 3
+sessions, so we can create a list of all the possible combinations. However, we
+need to remember that for session *ses-03* there is no BOLD data.
 
 .. code-block:: python
 
-   def get_elements(self):
+   from itertools import product
```
Comment (fraimondo) [inline docs/extending/datagrabber.rst:314]: Don't do this. Keep the example simple.

File: docs/extending/datagrabber.rst [hunk 2]
```diff
@@ -272,152 +272,172 @@ And we can create our datagrabber:
...
          sessions = ["ses-01", "ses-02"]
 
          # If we are not working on BOLD data, we can add "ses-03"
          if "BOLD" not in self.types:
             sessions.append("ses-03")
          elements = []
-         for subject in subjects:
-            for session in sessions:
-               elements.append({"subject": subject, "session": session})
+         for subject, element in product(subjects, sessions):
```
Comment (fraimondo) [inline docs/extending/datagrabber.rst:369]: same, see above

File: docs/extending/datagrabber.rst [hunk 3]
```diff
@@ -272,152 +272,172 @@ And we can create our datagrabber:
...
-            "variable1": "rot_x",
-            "variable2": "rot_z",
-            "variable3": "rot_y",
-         }
-      },
-   }
-
+   def get_item(
+      self, subject: str, session: str
+   ) -> dict[str, str | dict[str, str | dict[str, dict[str, str]]]]:
```
Comment (fraimondo) [inline docs/extending/datagrabber.rst:422]: I removed the typing from the examples on purpose.

While it's better for the actual code, it just makes the example more complicated to read with this kind of things.

**Review by fraimondo — CHANGES_REQUESTED** [review]

File: docs/extending/datagrabber.rst (hunk 3 above)
Comment (synchon) [reply docs/extending/datagrabber.rst:422]: I'll simplify it.

**Pull request merged**